nexus/src/nexus.rs

```diff
@@ -745,16 +766,12 @@ impl Nexus {
         project_name: &Name,
         params: &params::DiskCreate,
     ) -> CreateResult<db::model::Disk> {
-        let authz_project = self
-            .db_datastore
-            .project_lookup_by_path(organization_name, project_name)
+        let (.., authz_project) = LookupPath::new(opctx, &self.db_datastore)
+            .organization_name(organization_name)
+            .project_name(project_name)
+            .lookup_for(authz::Action::CreateChild)
             .await?;
 
-        // TODO-security This may need to be revisited once we implement authz
```

Comment:
This TODO is no longer applicable.  We did implement authz for saga actions.  The pattern is: we do the authz check before creating the saga for a better user experience, and we also serialize the current authn context, rehydrate it inside the saga, and use that to check again when we run the saga action.  So this TODO is resolved.

I suspect we're not actually authorizing the operations that this saga is doing, but that'll be caught when I sweep through DataStore looking for unprotected functions (which is tracked by #419).

The authorize check isn't removed -- it got moved into the lookup now.